srml/system/src/lib.rs

```diff
@@ -560,6 +577,9 @@ impl<T: Trait> Module<T> {
 	/// Remove temporary "environment" entries in storage.
 	pub fn finalize() -> T::Header {
 		ExtrinsicCount::kill();
+		// update the multiplier based on block weight.
+		NextFeeMultiplier::put(T::FeeMultiplierUpdate::convert(Self::all_extrinsics_weight()));
```

Comment:
ok I see if the updates are stateful over time (discussion above) then indeed this should be 
```diff
+ let current_weight = Self::all_extrinsics_weight();
+ let curren_multiplier = NextFeeMultiplier::get().apply_to(current_weight);
+ NextFeeMultiplier::put(T::FeeMultiplierUpdate::convert(current_multiplier));
- NextFeeMultiplier::put(T::FeeMultiplierUpdate::convert(Self::all_extrinsics_weight()));
```

Reply:
but that could be dangerous no? if we have low traffic (always less weight than ideal) then the fee keeps getting reduced and reduced and reduced...

Reply:
Maybe but at the same time if there low traffic during too much time it might means the offer is too expansive for the demand thus it make sense to reduce fee even more.

it is a `Slow adjusting mechanism`

Reply:
> ok I see if the updates are stateful over time (discussion above) then indeed this should be
> 
> ```diff
> + let current_weight = Self::all_extrinsics_weight();
> + let curren_multiplier = NextFeeMultiplier::get().apply_to(current_weight);
> + NextFeeMultiplier::put(T::FeeMultiplierUpdate::convert(current_multiplier));
> - NextFeeMultiplier::put(T::FeeMultiplierUpdate::convert(Self::all_extrinsics_weight()));
> ```

I'm not sure about this definition: new fee multiplier is a function of old fee multiplier and current block weight no ?

Reply:
> I'm not sure about this definition: new fee multiplier is a function of old fee multiplier and current block weight no ?

Yes exactly.